app/components/InternalLink/index.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import type { LinkProps } from 'react-router-dom';
+
+import { ResetScrollKey } from 'app/hooks/useResetScroll';
+
+interface InternalLinkProps extends Omit<LinkProps, 'state'> {
+  state?: { [key: string]: any };
+}
+
+function InternalLink(props: InternalLinkProps) {
+  const resetScrollState = {
+    [ResetScrollKey]: true,
+  };
+
+  const { state } = props;
+
+  const propsWithState = {
+    ...props,
+    ...(typeof state === 'undefined' ? {
+      state: resetScrollState,
+    } : {
+      state: {
+        ...state,
+        ...resetScrollState,
+      },
```

Comment:
It would be less clunky to just define a `state` property, then the assignment values would depend on if `props.state` is undefined or not.